### PR TITLE
Feature: Build/Test/Release script

### PR DESCRIPTION
**Pull request by kristofferahl**
Decided to replace the examples Makefile with a simple bash script and add a couple of things like:

- A command for building the provider (cross platform compiled using goreleaser)
- A command for releasing the provider (using goreleaser)
- Commands for running tests, integration test as well as dockerized test for different terraform versions

Also made a couple of smaller changes to the examples and readme.

What do you guys think? @masutaka @rossmckelvie

### CODE REVIEW

**Review by rossmckelvie — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -45,10 +47,67 @@ data "healthchecksio_channel" "pagerduty" {
...
 $ terraform import healthchecksio_check.my_first_check 760ca858-576a-432b-8b1f-378049d7ce96
 ```
+
+## Development
+
+### Pre-requisites
+- [Go](https://golang.org/) 1.12 or later
+- [Terraform](https://www.terraform.io/) v0.12 or later
+- [Healthchecks.io](https://healthchecks.io/) account and an API key
+- [goreleaser](https://goreleaser.com/) 0.85.1 or later
```
Comment (masutaka) [inline README.md:64]: Also go-healthchecksio? ;-)

**Review by masutaka — APPROVED** [review]

**Comment by masutaka** [comment]
By the way, will you introduce CI?

File: README.md (hunk 1 above)
Comment (kristofferahl) [reply README.md:64]: True, but we'll handle these kinds of dependencies using go modules. ;)

**Comment by kristofferahl** [comment]
@masutaka I'll add an issue to remind us setting up CI.

**Pull request merged**